File: Day 4/day4.py
def create_xmas_tree(height: int, ornament: str) -> str:
  longitudes  = [2*n+1 for n in range(height)]
  ancho = longitudes[-1]
  res = ""
  for lon in longitudes:
    n_guion = (ancho - lon) // 2
    guiones = '_'*n_guion
    linea = guiones + ornament*lon + guiones
    res = res + linea + "\n"

  n_tronco = (ancho - 1) // 2
  tronco = '_'*n_tronco 
  linea_tronco = tronco + '#' + tronco

  res += linea_tronco + "\n"
  res += linea_tronco 
  return res 

File: Day 4/day4.ts
function createXmasTree(height: number, ornament: string): string {
    const longitudes = Array.from({length: height}, (_, n) => 2*n + 1);
    const ancho = longitudes[longitudes.length-1];

    let res = ""
    for (const long of longitudes){
        const nGuion = (ancho - long) / 2;
        const guiones = '-'.repeat(nGuion);
        const linea = guiones + ornament.repeat(long) + guiones;
        res += linea + "\n";
    }

    const nTronco = (ancho - 1) / 2;
    const tronco = '_'.repeat(nTronco);
    const lineaTronco = tronco + '#' + tronco

    res += lineaTronco + "\n";
    res += lineaTronco;
    return res
}